service/history/workflow/cache/cache.go

```diff
@@ -418,6 +435,21 @@ func (c *CacheImpl) validateWorkflowID(
 	return nil
 }
 
+func (c *cacheItem) CacheSize() int {
+	if sg, ok := c.ctx.(cache.SizeGetter); ok {
+		return sg.CacheSize()
+	}
+	return 0
+}
```

Comment:
Redirects the cache size request to the inner workflow context.

Note that the cast is needed since some tests pass a MockContext in that does not implement this interface.